recipes/pfr/all/conanfile.py

```diff
@@ -0,0 +1,65 @@
+from conans import ConanFile, tools
+from conans.errors import ConanInvalidConfiguration
+import os
+
+
+class LibnameConan(ConanFile):
+    name = "pfr"
+    description = "std::tuple like methods for user defined types without any macro or boilerplate code"
+    topics = ("conan", "boost", "pfr", "reflection", "magic_get")
+    url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/apolukhin/magic_get"
+    license = "BSL-1.0"
+    no_copy_source = True
+    settings = "os", "compiler", "build_type", "arch"
+    exports_sources = "patches/**"
+
+    @property
+    def _source_subfolder(self):
+        return "source_subfolder"
+
+    @property
+    def _minimum_cpp_standard(self):
+        return 14
+
+    @property
+    def _minimum_compilers_version(self):
+        return {
+            "apple-clang": "9.4",
+            "clang": "3.8",
+            "gcc": "5.5"
+        }
+
+    def configure(self):
+        if self.settings.get_safe("compiler.cppstd"):
+            tools.check_min_cppstd(self, self._minimum_cpp_standard)
+
+        compiler = self.settings.compiler
+        try:
+            min_version = self._minimum_compilers_version[str(compiler)]
+            if tools.Version(compiler.version) < min_version:
+                msg = (
+                    "{} requires C++{} features which are not supported by compiler {} {}."
+                ).format(self.name, self._minimum_cpp_standard, compiler, compiler.version)
+                raise ConanInvalidConfiguration(msg)
+        except KeyError:
+            msg = (
+                "{} recipe lacks information about the {} compiler, "
+                "support for the required C++{} features is assumed"
+            ).format(self.name, compiler, self._minimum_cpp_standard)
+            self.output.warn(msg)
+
+    def source(self):
+        tools.get(**self.conan_data["sources"][self.version][0])
+        extracted_dir = "magic_get-" + self.version
+        os.rename(extracted_dir, self._source_subfolder)
+        for patch in self.conan_data.get("patches", {}).get(self.version, []):
+            tools.patch(**patch)
```

Comment:
Should be done in `package()`

Reply:
In `build`, you mean :wink:

Reply:
there's no build for header only so usually it goes in package.... since we have different understandings there should be a hook if thats feasible!

Reply:
A header-only can have a build step: e.g. for generating the headers.
But it does not matter in this case: build or package.

Reply:
I'm not convinced at all that patching sources in package() method is the good solution, and it is not what CCI does currently: 

- https://github.com/conan-io/conan-center-index/blob/master/recipes/boost/all/conanfile.py#L180
- https://github.com/conan-io/conan-center-index/blob/master/recipes/glad/all/conanfile.py#L91
- https://github.com/conan-io/conan-center-index/blob/master/recipes/jwt-cpp/all/conanfile.py#L27
- https://github.com/conan-io/conan-center-index/blob/master/recipes/lua/all/conanfile.py#L25
- https://github.com/conan-io/conan-center-index/blob/master/recipes/openexr/2.3.0/conanfile.py#L41
- https://github.com/conan-io/conan-center-index/blob/master/recipes/tgbot/all/conanfile.py#L33
- https://github.com/conan-io/conan-center-index/blob/master/recipes/tinyexr/all/conanfile.py#L50
- https://github.com/conan-io/conan-center-index/blob/master/recipes/websocketpp/all/conanfile.py#L34

I could patch it in build(), but it would require removing the no_copy_source attribute just for this purpose, I will do it if you both agree.

Reply:
Ouf.... https://github.com/conan-io/conan-center-index/pull/3123#discussion_r502468992 this is a recent comment from the CCI on the subject... but yes we have not been consistent